samza-core/src/main/java/org/apache/samza/operators/impl/StreamTableJoinOperatorImpl.java

```diff
@@ -57,10 +57,10 @@ protected void handleInit(Config config, TaskContext context) {
   }
 
   @Override
-  public Collection<JM> handleMessage(KV<K, MV> message, MessageCollector collector, TaskCoordinator coordinator) {
-    K key = message.getKey();
-    RV recordValue = table.get(key);
-    KV<K, RV> record = recordValue != null ? KV.of(key, recordValue) : null;
+  public Collection<JM> handleMessage(M message, MessageCollector collector, TaskCoordinator coordinator) {
+    K key = ((KV<K, ?>) message).getKey();
+    Object recordValue = table.get(key);
+    R record = recordValue != null ? (R) KV.of(key, recordValue) : null;
```

Comment:
It seems like that we still need to mandate R extends KV. I am a bit lost in this revision. I thought the goal was trying to get rid of K from the input type parameter in MessageStream.join(table, joinFn)? If we can not remove K from the type parameter, I think the version before this revision looks better.